src/component/TooltipBoundingBox.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { CSSProperties, PureComponent, ReactNode } from 'react';
+import { translateStyle } from 'react-smooth';
+import { AllowInDimension, AnimationDuration, AnimationTiming, CartesianViewBox, Coordinate } from '../util/types';
+import { getTooltipTranslate } from '../util/tooltip/translate';
+
+export type TooltipBoundingBoxProps = {
+  active: boolean;
+  allowEscapeViewBox: AllowInDimension;
+  animationDuration: AnimationDuration;
+  animationEasing: AnimationTiming;
+  children: ReactNode;
+  coordinate: Partial<Coordinate>;
+  hasPayload: boolean;
+  isAnimationActive: boolean;
+  offset: number;
+  position: Partial<Coordinate>;
+  reverseDirection: AllowInDimension;
+  useTranslate3d: boolean;
+  viewBox: CartesianViewBox;
+  wrapperStyle: CSSProperties;
+};
+
+type State = {
+  dismissed: boolean;
+  dismissedAtCoordinate: Coordinate;
+};
+
+const EPS = 1;
+
+export class TooltipBoundingBox extends PureComponent<TooltipBoundingBoxProps, State> {
+  state = {
+    dismissed: false,
+    dismissedAtCoordinate: { x: 0, y: 0 },
+  };
+
+  lastBoundingBox = {
+    width: -1,
+    height: -1,
+  };
+
+  private wrapperNode: HTMLDivElement;
+
+  updateBBox() {
+    if (this.wrapperNode && this.wrapperNode.getBoundingClientRect) {
+      const box = this.wrapperNode.getBoundingClientRect();
+
+      if (
+        Math.abs(box.width - this.lastBoundingBox.width) > EPS ||
+        Math.abs(box.height - this.lastBoundingBox.height) > EPS
+      ) {
+        this.lastBoundingBox.width = box.width;
```

Comment:
I think this instance assignment is a potential bug but I copy pasted it without a change.

Reply:
If you think this is a bug, please do elaborate, and create a follow up issue. We can tackle this independently. I appreciate this you only copied it in this PR.

Reply:
I think @HHongSeungWoo recently changed modified this section?

Reply:
@HHongSeungWoo confirmed in another thread that this is intentional: https://github.com/recharts/recharts/pull/3852#issuecomment-1806724245

I do not have any evidence that this is a bug. It's only an antipattern that can lead to bugs - this thing here is why `useState` and `useRef` exists.

Reply:
(Reading `defaultProps` and cloning elements are also antipatterns 😬 )

Reply:
This should probably be a function component then no? 😅

Reply:
While we have many anti-patterns, I think this one is similar to useMemo.
Due to `getTooltipTranslate`, this seems to be the best option for now.